relay-metrics/src/aggregation.rs

```diff
@@ -1824,49 +1976,33 @@ impl fmt::Debug for Aggregator {
     }
 }
 
-impl Actor for Aggregator {
-    type Context = Context<Self>;
-
-    fn started(&mut self, ctx: &mut Self::Context) {
-        relay_log::info!("aggregator started");
-
-        // Subscribe to shutdown
-        Controller::subscribe(ctx.address());
-
-        // TODO: Consider a better approach than busy polling
-        ctx.run_interval(FLUSH_INTERVAL, |slf, _context| {
-            slf.try_flush();
-        });
-    }
-
-    fn stopped(&mut self, _ctx: &mut Self::Context) {
-        relay_log::info!("aggregator stopped");
-    }
-}
-
-impl Default for Aggregator {
-    fn default() -> Self {
-        unimplemented!("register with the SystemRegistry instead")
-    }
-}
+impl Service for AggregatorService {
+    type Interface = Aggregator;
 
-impl Supervised for Aggregator {}
+    fn spawn_handler(mut self, mut rx: relay_system::Receiver<Self::Interface>) {
+        tokio::spawn(async move {
+            let mut ticker = time::interval(FLUSH_INTERVAL);
+            let mut shutdown = Controller::subscribe_v2().await;
+            relay_log::info!("AggregatorService started.");
+            loop {
+                tokio::select! {
+                    biased;
 
-impl SystemService for Aggregator {}
 
-impl Handler<Shutdown> for Aggregator {
-    type Result = Result<(), ()>;
+                    // NOTE: `compat::send_to_recipient` is required by ProjectCache and will be removed with ProjectCache migration.
+                    _ = ticker.tick() => self.try_flush(|recipient, flush_buckets| Box::pin(compat::send_to_recipient(recipient, flush_buckets))).await,
+                    Some(message) = rx.recv() => self.handle_message(message).await,
+                    _ = shutdown.changed() => self.handle_shutdown(&shutdown.borrow()),
 
-    fn handle(&mut self, message: Shutdown, _context: &mut Self::Context) -> Self::Result {
-        if message.timeout.is_some() {
-            relay_log::trace!("Shutting down...");
-            self.state = AggregatorState::ShuttingDown;
-        }
-        Ok(())
+                    else => break,
+                }
+            }
+            relay_log::info!("AggregatorService shutted down.");
```

Comment:
The old messages were "aggregator started" and "aggregator stopped":

```suggestion
            relay_log::info!("aggregator stopped");
```

I'm happy with a new message, however see spelling:
```suggestion
            relay_log::info!("AggregatorService shut down.");
```